internal/handlers/pg/pgdb/settings.go

```diff
@@ -0,0 +1,223 @@
+	var err error
+
+	schemaExists, err := pgPool.schemaExists(ctx, db)
+	if err != nil {
+		return "", lazyerrors.Error(err)
+	}
+
+	if !schemaExists {
+		return GetTableNameFormatted(collection), nil
+	}
+
+	var tables []string
+	if tables, err = pgPool.tables(ctx, db); err != nil {
+		return "", lazyerrors.Error(err)
+	}
+	if !slices.Contains(tables, collectionPrefix+"settings") {
+		err = pgPool.CreateSettingsTable(ctx, db)
+		if err != nil {
+			return "", lazyerrors.Error(err)
+		}
+	}
+
+	tx, err := pgPool.Begin(ctx)
+	if err != nil {
+		return "", lazyerrors.Error(err)
+	}
+	defer func() {
+		if err != nil {
+			must.NoError(tx.Rollback(ctx))
+			return
+		}
+		must.NoError(tx.Commit(ctx))
+	}()
+
+	settings, err := pgPool.getSettingsTable(ctx, tx, db)
+	if err != nil {
+		return "", lazyerrors.Error(err)
+	}
+
+	collectionsDoc := must.NotFail(settings.Get("collections"))
+	collections, ok := collectionsDoc.(*types.Document)
+	if !ok {
+		return "", lazyerrors.Errorf("invalid collections document: %v", collectionsDoc)
+	}
+
+	if collections.Has(collection) {
+		return must.NotFail(collections.Get(collection)).(string), nil
+	}
+
+	tableName := GetTableNameFormatted(collection)
+	must.NoError(collections.Set(collection, tableName))
+	must.NoError(settings.Set("collections", collections))
+
+	err = pgPool.updateSettingsTable(ctx, tx, db, settings)
+	if err != nil {
+		return "", lazyerrors.Error(err)
+	}
+
+	return tableName, nil
+}
+
+// updateSettingsTable updates FerretDB settings table.
+func (pgPool *Pool) updateSettingsTable(ctx context.Context, tx pgx.Tx, db string, settings *types.Document) error {
+	sql := `UPDATE ` + pgx.Identifier{db, collectionPrefix + "settings"}.Sanitize() + `SET settings = $1`
+	_, err := tx.Exec(ctx, sql, must.NotFail(fjson.Marshal(settings)))
+	return err
+}
+
+// getSettingsTable returns FerretDB settings table.
+func (pgPool *Pool) getSettingsTable(ctx context.Context, tx pgx.Tx, db string) (*types.Document, error) {
+	sql := `SELECT settings FROM ` + pgx.Identifier{db, collectionPrefix + "settings"}.Sanitize()
+	rows, err := tx.Query(ctx, sql)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+	defer rows.Close()
+
+	if !rows.Next() {
+		return nil, lazyerrors.Errorf("no settings found")
+	}
+
+	var b []byte
+	if err := rows.Scan(&b); err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	doc, err := fjson.Unmarshal(b)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	settings, ok := doc.(*types.Document)
+	if !ok {
+		return nil, lazyerrors.Errorf("invalid settings document: %v", doc)
+	}
+
+	return settings, nil
+}
+
+// RemoveTableFromSettings removes collection from FerretDB settings table.
+func (pgPool *Pool) RemoveTableFromSettings(ctx context.Context, db, collection string) error {
+	schemaExists, err := pgPool.schemaExists(ctx, db)
+	if err != nil {
+		return lazyerrors.Error(err)
+	}
+
+	if !schemaExists {
+		return nil
```

Comment:
Is here should be an error `ns not found`?

Reply:
In that particular case we should just exit.